runtime/startup_byt.c

```diff
@@ -566,18 +378,17 @@ CAMLexport void caml_main(char_os **argv)
   /* Load the globals */
   caml_seek_section(fd, &trail, "DATA");
   chan = caml_open_descriptor_in(fd);
-  Lock(chan);
-  caml_global_data = caml_input_val(chan);
-  Unlock(chan);
+  /* TODO: do we need multicore Lock here */
+  caml_modify_generational_global_root(&caml_global_data, caml_input_val(chan));
```

Comment:
Is there a reason for why `caml_global_data` needs to be handled differently in multicore?

Reply:
Tracking this for post-merge fixing in https://github.com/ocaml/ocaml/issues/10861